WebUI/src/App.vue

```diff
@@ -66,17 +66,19 @@
         @show-download-model-confirm="showDownloadModelConfirm"></create>
       <enhance v-show="activeTabIdx == 1" ref="enhanceCompt" @show-download-model-confirm="showDownloadModelConfirm">
       </enhance>
-      <answer v-show="activeTabIdx == 2" @show-download-model-confirm="showDownloadModelConfirm"></answer>
+      <answer v-show="activeTabIdx == 2" ref = "answer" @show-download-model-confirm="showDownloadModelConfirm" @show-model-request="showModelRequest"></answer>
       <learn-more v-show="activeTabIdx == 3"></learn-more>
       <app-settings v-if="showSetting" @close="hideAppSettings" @show-download-model-confirm="showDownloadModelConfirm"></app-settings>
     </div>
     <download-dialog v-show="showDowloadDlg" ref="downloadDigCompt" @close="showDowloadDlg = false"></download-dialog>
+    <add-l-l-m-dialog v-show="showModelRequestDialog" ref="addLLMCompt" @close="showModelRequestDialog = false" @call-check-model="callCheckModel" @show-warning="showWarning"></add-l-l-m-dialog>
+    <warning-dialog v-show="showWarningDialog" ref="warningCompt" @close="showWarningDialog = false"></warning-dialog>
```

Comment:
I am not yet convinced, that the llmDialog and the warning dialog should be on this level.  My feeling is: keep app.vue as slim as possible, otherwise we will not be able to mentally control it in the long run. We already have some "noise" of events, that simply flip booleans to display some popup/dialogs.

my gut feeling: introduce the add$ModelType as a component on its own , where type is probably (on the long run) corresponding to the view $answer/create/enhance -> llm/stablediff/??? . Probably, in considering the correct hierarchies, one arrives at a cleaner communication between components overall, especially when it comes to delegating events to the download component.


I would also be willing with this design for now and get feedback of intel. But imo we need to be very careful in not letting this degrade.


 I am however happy to hear, if you spent thoughts on this